substrate/frame/recovery/src/lib.rs

```diff
@@ -637,13 +661,16 @@ pub mod pallet {
 			// Take the active recovery process started by the rescuer for this account.
 			let active_recovery =
 				<ActiveRecoveries<T>>::take(&who, &rescuer).ok_or(Error::<T>::NotStarted)?;
-			// Move the reserved funds from the rescuer to the rescued account.
+			// Move the held funds from the rescuer to the rescued account.
 			// Acts like a slashing mechanism for those who try to maliciously recover accounts.
-			let res = T::Currency::repatriate_reserved(
+			let res = T::Currency::transfer_on_hold(
```

Comment:
I am thinking about how to do a lazy migration here. Maybe you can require the `Currency` config trait to be `Currency + MutateHold` so that it requires currency and fungibles.  

Then in this function you can first try to use the new `transfer_on_hold` function, but if that does not work then fallback to the old `repatriate_reserved`. Do you think this could work?  
Otherwise the MBMs are still in review and will need audit, so its a bit out.

Reply:
I am not a big fan of lazy migrations here, cause that will generate some "garbage" code that will live there forever. It seems more like a patch to me than a real solution. It's been a while already since we wanted the traits to be migrated that I would rather wait for the MBM to be ready and apply a proper solution